dbt/adapters/snowflake/relation.py

```diff
@@ -1,30 +1,41 @@
+import textwrap
+
 from dataclasses import dataclass, field
-from typing import FrozenSet, Optional, Type
+from typing import FrozenSet, Optional, Type, TYPE_CHECKING
+
 
 from dbt.adapters.base.relation import BaseRelation
 from dbt.adapters.contracts.relation import ComponentName, RelationConfig
+from dbt.adapters.events.types import AdapterEventWarning
 from dbt.adapters.relation_configs import (
     RelationConfigBase,
     RelationConfigChangeAction,
     RelationResults,
 )
 from dbt.adapters.utils import classproperty
 from dbt_common.exceptions import DbtRuntimeError
+from dbt_common.events.functions import warn_or_error
 
 from dbt.adapters.snowflake.relation_configs import (
     SnowflakeDynamicTableConfig,
     SnowflakeDynamicTableConfigChangeset,
     SnowflakeDynamicTableRefreshModeConfigChange,
     SnowflakeDynamicTableTargetLagConfigChange,
     SnowflakeDynamicTableWarehouseConfigChange,
+    SnowflakeObjectFormat,
     SnowflakeQuotePolicy,
     SnowflakeRelationType,
 )
 
+if TYPE_CHECKING:
+    from dbt.artifacts.resources.v1.model import ModelConfig
```

Comment:
Hopefully I'm being clever enough here. If I'm not, happy to remove this and keep the type hints as strings.

Reply:
Instead of using the ModelConfig from dbt-core we should rely on the RelationConfig protocol: https://github.com/dbt-labs/dbt-adapters/blob/main/dbt/adapters/contracts/relation.py#L49

Reply:
Done and done. Thanks for the pointer!